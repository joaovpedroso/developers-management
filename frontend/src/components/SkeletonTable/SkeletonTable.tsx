import { Grid, Skeleton } from "@mui/material";

const CustomSkeleton = ({
  width,
  height,
}: {
  width: string;
  height: string;
}) => (
  <Skeleton
    variant="rectangular"
    sx={{ bgcolor: "grey.900" }}
    width={width}
    height={height}
  />
);

const SkeletonTableRow = () => {
  return (
    <Grid
      sx={{
        margin: "2px 0 2px 0",
      }}
    >
      <CustomSkeleton width="100%" height="70px" />
    </Grid>
  );
};

const SkeletonPagination = () => (
  <Grid
    sx={{
      marginTop: "4px",
    }}
  >
    <CustomSkeleton width="100%" height="50px" />
  </Grid>
);

const SkeletonTable = () => {
  return (
    <Grid width="100%" marginTop="24px">
      <CustomSkeleton width="100%" height="50px" />
      <SkeletonTableRow />
      <SkeletonTableRow />
      <SkeletonTableRow />
      <SkeletonTableRow />
      <SkeletonTableRow />
      <SkeletonTableRow />
      <SkeletonPagination />
    </Grid>
  );
};

export { SkeletonTable };
