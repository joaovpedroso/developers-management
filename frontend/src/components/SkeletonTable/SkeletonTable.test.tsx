import { render } from "@testing-library/react";
import { SkeletonTable } from "./SkeletonTable";

describe("SkeletonTable", () => {
  it("should match snapshot", () => {
    const { container } = render(<SkeletonTable />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
