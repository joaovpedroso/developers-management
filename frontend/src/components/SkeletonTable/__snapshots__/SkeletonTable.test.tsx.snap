// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonTable should match snapshot 1`] = `
<div
  class="MuiGrid-root css-10xli1p-MuiGrid-root"
>
  <span
    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-16fe59p-MuiSkeleton-root"
    style="width: 100%; height: 50px;"
  />
  <div
    class="MuiGrid-root css-1h5myfb-MuiGrid-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-16fe59p-MuiSkeleton-root"
      style="width: 100%; height: 70px;"
    />
  </div>
  <div
    class="MuiGrid-root css-1h5myfb-MuiGrid-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-16fe59p-MuiSkeleton-root"
      style="width: 100%; height: 70px;"
    />
  </div>
  <div
    class="MuiGrid-root css-1h5myfb-MuiGrid-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-16fe59p-MuiSkeleton-root"
      style="width: 100%; height: 70px;"
    />
  </div>
  <div
    class="MuiGrid-root css-1h5myfb-MuiGrid-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-16fe59p-MuiSkeleton-root"
      style="width: 100%; height: 70px;"
    />
  </div>
  <div
    class="MuiGrid-root css-1h5myfb-MuiGrid-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-16fe59p-MuiSkeleton-root"
      style="width: 100%; height: 70px;"
    />
  </div>
  <div
    class="MuiGrid-root css-1h5myfb-MuiGrid-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-16fe59p-MuiSkeleton-root"
      style="width: 100%; height: 70px;"
    />
  </div>
  <div
    class="MuiGrid-root css-1nogpbe-MuiGrid-root"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-16fe59p-MuiSkeleton-root"
      style="width: 100%; height: 50px;"
    />
  </div>
</div>
`;
